### PR TITLE
[jasmine] Use type-safe spies for spyOnProperty

### PR DESCRIPTION
**Pull request by rslawik**
Same as #59463 but for v3.

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://jasmine.github.io/api/edge/global.html#spyOnProperty

### CODE REVIEW

**Review by chivesrs — APPROVED** [review]

**Comment by rslawik** [comment]
@chivesrs Do you know who else needs to approve this PR? It is effectively a backport of #59463 to v3/, so it should be a quick review.

**Comment by chivesrs** [comment]
You need to wait for a DT (this repository) maintainer.

File: types/jasmine/v3/index.d.ts [hunk 1]
```diff
@@ -191,7 +191,10 @@ declare function spyOn<T, K extends keyof T = keyof T>(
  * @param property The name of the property to replace with a `Spy`.
  * @param accessType The access type (get|set) of the property to `Spy` on.
  */
-declare function spyOnProperty<T>(object: T, property: keyof T, accessType?: "get" | "set"): jasmine.Spy;
+ declare function spyOnProperty<T, K extends keyof T = keyof T>(
```
Comment (rbuckton) [inline types/jasmine/v3/index.d.ts:194]: nit: indentation

```suggestion
declare function spyOnProperty<T, K extends keyof T = keyof T>(
```

**Review by rbuckton — CHANGES_REQUESTED** [review]

**Review by chivesrs — APPROVED** [review]

**Review by jakebailey — APPROVED** [review]

**Comment by rslawik** [comment]
Ready to merge

**Pull request merged**